examples/go-app-counter/main.go

```diff
@@ -0,0 +1,324 @@
+						ProgramType: *bpfmanHelpers.Tracepoint.Uint32(),
+						Attach: &gobpfman.AttachInfo{
+							Info: &gobpfman.AttachInfo_TracepointAttachInfo{
+								TracepointAttachInfo: &gobpfman.TracepointAttachInfo{
+									Tracepoint: "syscalls/sys_enter_kill",
+								},
+							},
+						},
+					}
+				}
+			case configMgmt.ProgTypeUprobe:
+				fnName := "malloc"
+				if paramData.MapOwnerId != 0 {
+					mapOwnerId := uint32(paramData.MapOwnerId)
+					loadRequest = &gobpfman.LoadRequest{
+						Bytecode:    paramData.BytecodeSource,
+						Name:        "uprobe_counter",
+						ProgramType: *bpfmanHelpers.Kprobe.Uint32(),
+						Attach: &gobpfman.AttachInfo{
+							Info: &gobpfman.AttachInfo_UprobeAttachInfo{
+								UprobeAttachInfo: &gobpfman.UprobeAttachInfo{
+									FnName: &fnName,
+									Target: "libc",
+								},
+							},
+						},
+						MapOwnerId: &mapOwnerId,
+					}
+				} else {
+					loadRequest = &gobpfman.LoadRequest{
+						Bytecode:    paramData.BytecodeSource,
+						Name:        "uprobe_counter",
+						ProgramType: *bpfmanHelpers.Kprobe.Uint32(),
+						Attach: &gobpfman.AttachInfo{
+							Info: &gobpfman.AttachInfo_UprobeAttachInfo{
+								UprobeAttachInfo: &gobpfman.UprobeAttachInfo{
+									FnName: &fnName,
+									Target: "libc",
+								},
+							},
+						},
+					}
+				}
+			case configMgmt.ProgTypeXdp:
+				if paramData.MapOwnerId != 0 {
+					mapOwnerId := uint32(paramData.MapOwnerId)
+					loadRequest = &gobpfman.LoadRequest{
+						Bytecode:    paramData.BytecodeSource,
+						Name:        "xdp_stats",
+						ProgramType: *bpfmanHelpers.Xdp.Uint32(),
+						Attach: &gobpfman.AttachInfo{
+							Info: &gobpfman.AttachInfo_XdpAttachInfo{
+								XdpAttachInfo: &gobpfman.XDPAttachInfo{
+									Priority: int32(paramData.Priority),
+									Iface:    paramData.Iface,
+								},
+							},
+						},
+						MapOwnerId: &mapOwnerId,
+					}
+				} else {
+					loadRequest = &gobpfman.LoadRequest{
+						Bytecode:    paramData.BytecodeSource,
+						Name:        "xdp_stats",
+						ProgramType: *bpfmanHelpers.Xdp.Uint32(),
+						Attach: &gobpfman.AttachInfo{
+							Info: &gobpfman.AttachInfo_XdpAttachInfo{
+								XdpAttachInfo: &gobpfman.XDPAttachInfo{
+									Priority: int32(paramData.Priority),
+									Iface:    paramData.Iface,
+								},
+							},
+						},
+					}
+				}
+			}
+
+			// 1. Load Program using bpfman
+			var res *gobpfman.LoadResponse
+			res, err = c.Load(ctx, loadRequest)
+			if err != nil {
+				conn.Close()
+				log.Print(err)
+				return nil, action, err
+			}
+
+			kernelInfo := res.GetKernelInfo()
+			if kernelInfo != nil {
+				paramData.ProgId = uint(kernelInfo.GetId())
+			} else {
+				conn.Close()
+				log.Printf("kernelInfo not returned in LoadResponse")
+				return nil, action, err
+			}
+			log.Printf("Program registered with id %d\n", paramData.ProgId)
+
+			// 2. Set up defer to unload program when this is closed
+			defer func(id uint) {
```

Comment:
Having this here unloads the program as soon as this function exits, so it doesn't work.  The deferred close needs to be in the function that runs the get stats loop or after it exits.